src/she/sdl2/sdl2_surface.cpp

```diff
@@ -422,7 +423,17 @@ namespace she {
   void SDL2Surface::fillRect(gfx::Color color, const gfx::Rect& rc)
   {
     SDL_Rect rect{rc.x, rc.y, rc.w, rc.h};
-    SDL_FillRect(m_bmp, &rect, to_sdl(m_bmp->format, color));
+    if (gfx::is_solid(color)){
+        SDL_FillRect(m_bmp, &rect, to_sdl(m_bmp->format, color));
+    } else {
+        // Make temporary surface with the color and blit it with alpha-blend
+        SDL_Surface *surface_tmp = SDL_CreateRGBSurface(0, rc.w, rc.h, 32, 0xFF, 0xFF00, 0xFF0000, 0xFF000000);
```

Comment:
Is this the only way to do this? IIRC, fillRect can be used really frequently and creating/destroying a temporary surface all the time is really inefficient. Maybe keep the surface and only recreate it if the size changes?

Reply:
It's only doing this for transparent colors. Solid colors run the other piece of code and only use a fillrect.
I'm open to add optimizations, but this is the simplest way I found.

We can keep a surface and use a rectangle on the blit operation so we only replace the surface if the rectangle we need to repaint is larger than the surface. That is possible, too.